pkg/skills/loader.go

```
@@ -15,63 +15,95 @@ type SkillMeta struct {
+	if _, err := os.Stat(skillsDir); os.IsNotExist(err) {
+		return nil, nil
 	}
 
 	var skills []SkillMeta
-	for _, entry := range entries {
-		if !entry.IsDir() {
-			continue
+
+	err := filepath.WalkDir(skillsDir, func(path string, d os.DirEntry, err error) error {
+		if err != nil {
+			return nil
```

golangci-lint (2.12.2)

[error] error is not nil (line 27) but it returns nil

(nilerr)